PLFM-8519 - Add `userHasDownloadPermission` to `RestrictionInformationResponse`

nickgros opened the pull request.

nickgros committed:
PLFM-8519 - Add `userHasDownloadPermission` to `RestrictionInformationResponse`

john-hill reviewed (changes requested):
Can we explore refactoring the decider logic into a static util that can be used for both the mangers?

nickgros committed:
PLFM-8519 - refactor from code review

- Modify RestrictionInformationResponse to include a `userEntityPermissions` field, remove `isDataContributor` field
- Update UserEntityPermissions to include `isDataContributor` field
- Refactor so functions to create UserEntityPermissions objects are static utilities
- Add ability to pass existing EntityStateProvider to EntityAuthorizationManager.getUserPermissionsForEntity
- Use existing EntityStateProvider to calculate permissions in RestrictionInformationManagerImpl

nickgros committed:
PLFM-8519 - fix integration test

marcomarasca reviewed (approved):
LGTM!
